tests/integrations/strawberry/test_strawberry_py3.py

```diff
@@ -0,0 +1,894 @@
+
+    assert len(events) == 1
+
+    (error_event,) = events
+    assert error_event["exception"]["values"][0]["mechanism"]["type"] == "strawberry"
+    assert "data" not in error_event["request"]
+    assert "response" not in error_event["contexts"]
+
+    assert len(error_event["breadcrumbs"]["values"]) == 1
+    assert error_event["breadcrumbs"]["values"][0]["category"] == "graphql.operation"
+    assert error_event["breadcrumbs"]["values"][0]["data"] == {
+        "operation_name": "ErrorQuery",
+        "operation_type": "query",
+    }
+
+
+def test_breadcrumb_no_operation_name_async(sentry_init, capture_events):
+    sentry_init(
+        integrations=[
+            StrawberryIntegration(),
+            FastApiIntegration(),
+            StarletteIntegration(),
+        ],
+    )
+    events = capture_events()
+
+    schema = strawberry.Schema(Query)
+
+    async_app = FastAPI()
+    async_app.include_router(GraphQLRouter(schema), prefix="/graphql")
+
+    query = "{ error }"
+    client = TestClient(async_app)
+    client.post("/graphql", json={"query": query})
+
+    assert len(events) == 1
+
+    (error_event,) = events
+
+    assert len(error_event["breadcrumbs"]["values"]) == 1
+    assert error_event["breadcrumbs"]["values"][0]["category"] == "graphql.operation"
+    assert error_event["breadcrumbs"]["values"][0]["data"] == {
+        "operation_name": None,
+        "operation_type": "query",
+    }
+
+
+def test_breadcrumb_no_operation_name_sync(sentry_init, capture_events):
+    sentry_init(
+        integrations=[StrawberryIntegration(async_execution=False), FlaskIntegration()],
+    )
+    events = capture_events()
+
+    schema = strawberry.Schema(Query)
+
+    sync_app = Flask(__name__)
+    sync_app.add_url_rule(
+        "/graphql",
+        view_func=GraphQLView.as_view("graphql_view", schema=schema),
+    )
+
+    query = "{ error }"
+    client = sync_app.test_client()
+    client.post("/graphql", json={"query": query})
+
+    assert len(events) == 1
+
+    (error_event,) = events
+
+    assert len(error_event["breadcrumbs"]["values"]) == 1
+    assert error_event["breadcrumbs"]["values"][0]["category"] == "graphql.operation"
+    assert error_event["breadcrumbs"]["values"][0]["data"] == {
+        "operation_name": None,
+        "operation_type": "query",
+    }
+
+
+def test_capture_transaction_on_error_async(sentry_init, capture_events):
+    sentry_init(
+        send_default_pii=True,
+        integrations=[
+            StrawberryIntegration(),
+            FastApiIntegration(),
+            StarletteIntegration(),
+        ],
+        traces_sample_rate=1,
+    )
+    events = capture_events()
+
+    schema = strawberry.Schema(Query)
+
+    async_app = FastAPI()
+    async_app.include_router(GraphQLRouter(schema), prefix="/graphql")
+
+    query = "query ErrorQuery { error }"
+    client = TestClient(async_app)
+    client.post("/graphql", json={"query": query, "operationName": "ErrorQuery"})
```

Comment:
Perhaps this logic (and the equivalent logic for the sync case) could be factored out into a fixture, so that we can use the same test functions for both sync and async and have each test parameterized by the logic for constructing the async and sync apps and queries

This comment also applies to most other tests in this file

Reply:
Updated to use fixtures!